<a href="https://colab.research.google.com/github/aparna-2001/machine_learning_using_python/blob/main/level1_ML_foundation_BCI.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Alphawaves data,

subject 0

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.io import loadmat


- numpy for numerical computing
- pandas for data handling
- matplotlib for plotting and visualising
- seaborn for visualising
- scipy, loadmat for loading the .mat files


In [3]:
#loading the .mat file in loadmat

data = loadmat('subject_00.mat')
print(data.keys())

print(type(data['SIGNAL']))

signal = data['SIGNAL']

if hasattr(signal, 'dtype') and signal.dtype.names:
    print(signal.dtype.names)


dict_keys(['__header__', '__version__', '__globals__', 'SIGNAL'])
<class 'numpy.ndarray'>


In [4]:
#extracting the signal only

signal_data = pd.DataFrame(data['SIGNAL'])

print(signal_data.shape)

signal_data.head(5)

(63104, 19)


0            1            2           3            4            5   \
0  0.000000  2314.710938 -2473.864258  304.726593 -6740.526367 -6507.790039   
1  0.001953  2312.077393 -2477.192871  295.271210 -6747.804199 -6508.391602   
2  0.003906  2308.139160 -2477.750000  292.862305 -6750.519043 -6510.719238   
3  0.005859  2303.175293 -2477.378174  296.507050 -6756.901856 -6515.480957   
4  0.007812  2307.331543 -2471.294434  301.831573 -6754.847656 -6509.622070   

             6            7             8            9            10  \
0 -11247.854492 -2859.524170 -14004.568359 -1706.199707 -4104.730469   
1 -11249.747070 -2863.340820 -14008.770508 -1708.395874 -4104.203613   
2 -11248.869141 -2867.504883 -14007.042969 -1709.802368 -4105.119141   
3 -11250.728516 -2872.019775 -14011.469727 -1709.260498 -4109.889648   
4 -11247.145508 -2865.405518 -14011.854492 -1702.394165 -4102.324707   

            11            12           13          14          15  \
0 -6290.920410 -12415.869141 -7290.736816  170.436508  397.110260   
1 -6288.968262 -12418.380859 -7293.556641  175.323837  400.196503   
2 -6274.008301 -12422.494141 -7294.418945  193.758057  410.232483   
3 -6266.696777 -12422.033203 -7302.818848  197.839844  410.054993   
4 -6257.812012 -12413.977539 -7298.030273  200.905090  415.010742   

            16   17   18  
0 -3278.888184  0.0  0.0  
1 -3276.674561  0.0  0.0  
2 -3263.133545  0.0  0.0  
3 -3260.899902  0.0  0.0  
4 -3255.357422  0.0  0.0

In [5]:
#signal_data.iloc[0:6, 0]

first_row = signal_data.iloc[0, 0]
second_row = signal_data.iloc[1, 0]
third_row = signal_data.iloc[2, 0]

print(second_row - first_row)
print(third_row - second_row)


0.001953125
0.001953125


In [6]:
#sampling rate

sampling_rate = 1/(signal_data.iloc[1,0] - signal_data.iloc[0, 0])

print(sampling_rate)

512.0


In [7]:
#recording time

recording_time = signal_data.iloc[-1,0] - signal_data.iloc[0,0]
print(recording_time)

123.24804688


In [8]:
#confirming the sample rate

sampling_rate_conf = signal_data.shape[0]/recording_time
print(sampling_rate_conf)

512.0081136980691


since the difference is uniform, the first column will be the time

- file is in .mat format.

- The data is structured as rows contain observation at each time.  

- sample and columns contains the recordings of EEG electrodes.

- sampling rate - total number of samples/ total recording time in seconds; or 1/(time[1] - time[0]) , sampling rate of this data is 512 Hz

- How long is the recording ; The recording time is 123.24804688 seconds, approximately close to 2 minutes


In [13]:
#shape of the data

print(signal_data.shape)

print(signal_data.columns,'\n')

print(signal_data.info())

(63104, 19)
RangeIndex(start=0, stop=19, step=1) 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63104 entries, 0 to 63103
Data columns (total 19 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       63104 non-null  float64
 1   1       63104 non-null  float64
 2   2       63104 non-null  float64
 3   3       63104 non-null  float64
 4   4       63104 non-null  float64
 5   5       63104 non-null  float64
 6   6       63104 non-null  float64
 7   7       63104 non-null  float64
 8   8       63104 non-null  float64
 9   9       63104 non-null  float64
 10  10      63104 non-null  float64
 11  11      63104 non-null  float64
 12  12      63104 non-null  float64
 13  13      63104 non-null  float64
 14  14      63104 non-null  float64
 15  15      63104 non-null  float64
 16  16      63104 non-null  float64
 17  17      63104 non-null  float64
 18  18      63104 non-null  float64
dtypes: float64(19)
memory usage: 9.1 MB
None


Answer these questions by exploring:

What is the shape of your data? (rows × columns) - The shape of the data is 63104 * 19

What are the column names? 0 - 19

Are there any missing values? (Use appropriate method)


What data type is each column? - each data type is float
